pkg/xds/experimental/client/cell.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/node"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	corepb "github.com/cilium/proxy/go/envoy/config/core/v3"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+	core_v1 "k8s.io/api/core/v1"
+)
+
+var Cell = cell.Module(
+	"xds-client",
+	"xDS client",
+
+	cell.Config(defaultConfig),
+	cell.Provide(NewXDSClient),
+)
+
+type Config struct {
+	ServerAddr string `mapstructure:"xds-server-address"`
+
+	Options
+}
+
+var defaultConfig = Config{
+	ServerAddr: "",
+	Options:    *Defaults,
+}
+
+func (conf Config) Flags(flags *pflag.FlagSet) {
+	flags.String("xds-server-address", conf.ServerAddr, "Address of xDS server")
+	flags.Bool("xds-use-sotw-protocol", conf.UseSOTW, "Use State Of The World, non-incremental version of xDS protocol")
+}
+
+type input struct {
+	cell.In
+
+	Config
+	JobGroup       job.Group
```

Comment:
It's kind of weird that you have unexpected struct with exported fields

Reply:
What is weird about it? (Cell, documentation, and previous review, suggests the current format)